### PR TITLE
Outline public transport

### PR DESCRIPTION
**Pull request by BlairArchibald**
This PR contains an outline of the public transport (as I understand we want it) for feedback. Still to add parameters etc if this is the approach we are going for.

The DailyStats test fails (but this seems to be unrelated and just an unlucky 0 asymptomatic when it finishes. But we can look into this if we are happy with the approach).

Not clear how to factor out the infection stuff to one class (other than to make Transport a special "Place"). But maybe having a little duplication here is okay.

Currently, a household either takes public transport or doesn't (just to avoid adults taking the train and leaving children to drive!).

For parameters more than 10% taking public transport things get pretty slow. But the new sample based infection approach should be O(n) and solve this I think.

### CODE REVIEW

**Comment by BlairArchibald** [comment]
I think this is now ready for a final review.

I've added public transport parameters, and with the new binomial sampling in place it's not too slow to do this even when lots of people are moving.

One outstanding question is if we should trace contacts on public transport @johnnonweiler ? For now we do, to keep Transport as close to a "place" as possible.

We also need realistic parameters, but this is probably best coming from @paulbessell in a separate commit.

**Comment by johnnonweiler** [comment]
I have pushed a commit to fix the merge conflicts.

**Comment by johnnonweiler** [comment]
I have now amended my merge commit (after I made a mistake, in not stashing other changes which then got accidentally committed).

**Comment by BlairArchibald** [comment]
Great, thanks!

Given than effects the epidemiology let's wait for @paulbessell before merging this

File: src/main/java/uk/co/ramp/covid/simulation/population/Transport.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import uk.co.ramp.covid.simulation.parameters.PopulationParameters;
+import uk.co.ramp.covid.simulation.place.Place;
+import uk.co.ramp.covid.simulation.util.Time;
+
+/** Transport is a special form of Place where we allow infections to occur, but not to advance,
+ *  and no movements occur */
+public class Transport extends Place {
+
+    public Transport() {
+        transAdjustment = PopulationParameters.get().publicTransportParameters.transmissionConstant;
```
Comment (paulbessell) [inline src/main/java/uk/co/ramp/covid/simulation/population/Transport.java:14]: Slightly concerned about this bit. Could we write an override for `Place.getTransConstant()`. In my view `Transport.getTransConstant()` should return 
`transConstant * transAdjustment / Population.populationSize`

The reason being that in its current formulation the Transport transmisison scales such that it is constant with the number of people that ar emoving at that time. In reality there will be more transmission when more people are using Transport. Does that make sense?

File: src/main/java/uk/co/ramp/covid/simulation/population/Transport.java [hunk 2]
```diff
@@ -23,6 +26,15 @@ public void addPassenger(Person p) {
...
     public void determineMovement(Time t, DailyStats s, boolean lockdown, Places places) { }
 
+    @Override
+    protected double getTransConstant() {
+        if(people.size() == 0) {
+            return 0.0;
+        }
+
+        return transConstant * transAdjustment / populationSize;
+    }
```
Comment (BlairArchibald) [inline src/main/java/uk/co/ramp/covid/simulation/population/Transport.java:35]: Is this now correct?

**Comment by paulbessell** [comment]
Looks great. Cheers Blair

**Pull request merged**